basic_block.c

```diff
@@ -5,6 +5,78 @@
 #include <stdio.h>
 #endif
 
+#if GUM_NATIVE_CPU == GUM_CPU_AMD64
+
+static const guint8 afl_maybe_log_code[] = {
+    0x9c,                                      /* pushfq */
+    0x50,                                      /* push rax */
+    0x51,                                      /* push rcx */
+    0x52,                                      /* push rdx */
+    0x56,                                      /* push rsi */
+
+    0x89, 0xf8,                                /* mov eax, edi */
+    0xc1, 0xe0, 0x08,                          /* shl eax, 8 */
+    0xc1, 0xef, 0x04,                          /* shr edi, 4 */
+    0x31, 0xc7,                                /* xor edi, eax */
+    0x0f, 0xb7, 0xc7,                          /* movzx eax, di */
+    0x48, 0x8d, 0x0d, 0x46, 0x00, 0x00, 0x00,  /* lea rcx, sym._afl_instr_rms */
+    0x8b, 0x09,                                /* mov ecx, dword [rcx] */
+    0x48, 0x39, 0xc8,                          /* cmp rax, rcx */
+    0x73, 0x29,                                /* jae beach */
+    0x48, 0x8d, 0x0d, 0x30, 0x00, 0x00, 0x00,  /* lea rcx, sym._afl_area_ptr_ptr */
+    0x48, 0x8b, 0x09,                          /* mov rcx, qword [rcx] */
+    0x48, 0x8b, 0x09,                          /* mov rcx, qword [rcx] */
+    0x48, 0x8d, 0x15, 0x1b, 0x00, 0x00, 0x00,  /* lea rdx, sym._afl_prev_loc_ptr */
+    0x48, 0x8b, 0x32,                          /* mov rsi, qword [rdx] */
+    0x48, 0x8b, 0x36,                          /* mov rsi, qword [rsi] */
+    0x48, 0x31, 0xc6,                          /* xor rsi, rax */
+    0xfe, 0x04, 0x31,                          /* inc byte [rcx + rsi] */
```

Comment:
state of the art is adding an addc afterwards to prevent wrapping to 0 but jumping to 1 instead

Reply:
Thanks, good point! We should do this in the C implementation and then update this to match. /cc @meme